src/Option/Option.ts

```diff
@@ -142,6 +142,29 @@ export const map = <A extends NonNullish, B extends NonNullish>(
         none: none,
     })
 
+/**
+ * Execute an arbitrary side-effect function if `Some`.
+ * Does not call the function for `None`.
+ *
+ * @group Utils
+ *
+ * @example
+ * pipe(
+ *     Option.some("cheese"),
+ *     Option.iter(s => fooCallback(s))
+ * ) // fooCallback is called with "cheese"
+ *
+ * pipe(
+ *     Option.none,
+ *     Option.iter((s: string) => fooCallback(s))
+ * ) // fooCallback is not called
+ */
+export const iter = <A extends NonNullish>(f: (a: A) => void) =>
+    match<A, void>({
+        some: a => f(a),
+        none: () => {},
```

Comment:
Let me know if there's a better way to do this such that we aren't creating a useless arrow function.

Reply:
See https://github.com/fp-toolkit/fp-toolkit/pull/72/files#r1735068930. If `void 0` doesn't work, try `undefined` as well